core/core_integrations/slack.js

```diff
@@ -17,596 +15,553 @@ inTransaction = {},
+        var message = messageQueue[teamId].shift();
+        request({
+                uri: message.uri,
+                method: 'GET',
+                qs: message.body
+            },
+            function (error, response, body) {
+                body = JSON.parse(body);
+                if (error) {
+                    console.debug('slack-> error: ' + error);
+                    message.callback(true, null);
+                } else if (!body.ok) {
+                    console.debug('slack-> Failed to send message, error: ' + body.error);
+                    message.callback(true, null);
+                } else if (message.callback) {
+                    message.callback(false, body);
+                }
+                inTransaction[teamId] = false;
+                // Wait until message is sent before trying to send another message.
+                sendMessageWebAPI(teamId);
+            }
+        );
+    }
 },
 
-sendMessageWebAPI = function (teamId) {
-	// Initialise for team
-	if (inTransaction[teamId] === null || inTransaction[teamId] === undefined) {
-		inTransaction[teamId] = false;
+addMessageToQueue = function(teamId, body, uri, callback) {
+    waitingForFirstTransaction[teamId] = false;
+    if (!messageQueue[teamId]) {
+        messageQueue[teamId] = [];
     }
-
-	if (messageQueue[teamId].length >= 1 && !inTransaction[teamId]) {
-	    inTransaction[teamId] = true;
-	    var message = messageQueue[teamId].shift();
-	    request({
-	            "uri": message.uri,
-	            "method": 'GET',
-	            "qs": message.body
-	        },
-	        function (error, response, body) {
-	            body = JSON.parse(body);
-	            if (error) {
-	                console.debug('slack-> error: ' + error);
-	                message.callback(true, null);
-	            } else if (!body.ok) {
-	                console.debug('slack-> Failed to send message, error: ' + body.error);
-	                message.callback(true, null);
-	            } else if (message.callback) {
-	                message.callback(false, body);
-	            }
-	            inTransaction[teamId] = false;
-	            // Wait until message is sent before trying to send another message.
-	            sendMessageWebAPI(teamId);
-	    	});
-	}
-}
+    messageQueue[teamId].push({body: body, uri: uri, callback: callback});
+    sendMessageWebAPI(teamId);
+},
 
 getChannelIdAndTeamId = function(thread) {
-	var slackTeams = exports.config.slack_teams,
-	thread_components = thread.split('\0', 2),
-	channelId = thread_components[0],
-	teamId = thread_components[1],
-	token = slackTeams[teamId].token;
+    var thread_components = thread.split('\0', 2),
+    channelId = thread_components[0],
+    teamId = thread_components[1],
+    token = teamData[teamId].token;
+
+    return { channel_id: channelId, team_id: teamId, token: token };
+},
 
-	return { "channel_id": channelId, "team_id": teamId, "token": token };
+sendMessage = function(message, thread) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l71)